### ISSUE TITLE
Build error target "Memory"

### ISSUE THREAD

**Issue by erksch** [facebookresearch/flashlight#96]
Hey guys sorry to bother you with yet another build error issue.
I freshly cloned the repo and executed the build for the CUDA backend with:

`cmake .. -DCMAKE_BUILD_TYPE=Release -DFLASHLIGHT_BACKEND=CUDA`

And I receive the following error: 

```
CMake Error at flashlight/memory/CMakeLists.txt:3 (add_library):
  add_library cannot create target "Memory" because another target with the
  same name already exists.  The existing target is an interface library
  created in source directory "/home/erik/Projects/flashlight".  See
  documentation for policy CMP0002 for more details.
```

<details>
<summary>Here's the full log.</summary>

```
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 9.2.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- ArrayFire found (include: /usr/local/arrayfire/include, library: ArrayFire::afcuda)
-- Could NOT find cereal (missing: cereal_INCLUDE_DIRS) 
-- cereal NOT found. Will download from source
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- CUDA found (library: /usr/local/cuda/lib64/libcudart_static.a;-lpthread;dl;/usr/lib/x86_64-linux-gnu/librt.so include: /usr/local/cuda/include)
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Found CUDNN: (lib: /usr/lib/x86_64-linux-gnu/libcudnn.so include: /usr/include)
-- CUDNN found (library: /usr/lib/x86_64-linux-gnu/libcudnn.so include: /usr/include)
-- Will build flashlight contrib assets.
CMake Warning (dev) at flashlight/contrib/modules/CMakeLists.txt:18 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "ContribModules" has a relative path.
Call Stack (most recent call first):
  flashlight/contrib/CMakeLists.txt:12 (include)
  CMakeLists.txt:219 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at flashlight/autograd/CMakeLists.txt:16 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "Autograd" has a relative path.
Call Stack (most recent call first):
  CMakeLists.txt:255 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at flashlight/autograd/CMakeLists.txt:79 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "Autograd" has a relative path.
Call Stack (most recent call first):
  CMakeLists.txt:255 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at flashlight/common/CMakeLists.txt:25 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "Common" has a relative path.
Call Stack (most recent call first):
  CMakeLists.txt:258 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
CMake Warning (dev) at flashlight/dataset/CMakeLists.txt:25 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "Dataset" has a relative path.
Call Stack (most recent call first):
  CMakeLists.txt:261 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at flashlight/distributed/CMakeLists.txt:19 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "Distributed" has a relative path.
Call Stack (most recent call first):
  CMakeLists.txt:264 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Gloo not found
-- Determining NCCL version from the header file: /usr/include/nccl.h
-- NCCL_MAJOR_VERSION: 2
-- Found NCCL (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libnccl.so)
-- NCCL found
-- MPI_CXX found
-- MPI_CXX compile flags: 
-- MPI_CXX include path: /usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/usr/lib/x86_64-linux-gnu/openmpi/include
-- MPI_CXX LINK flags path: -pthread
-- MPI_CXX libraries: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so/usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so
-- MPI_C found
-- MPI_C compile flags: 
-- MPI_C include path: /usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/usr/lib/x86_64-linux-gnu/openmpi/include
-- MPI_C LINK flags path: -pthread
-- MPI_C libraries: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so
CMake Warning (dev) at flashlight/memory/CMakeLists.txt:20 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "Memory" has a relative path.
Call Stack (most recent call first):
  CMakeLists.txt:267 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at flashlight/meter/CMakeLists.txt:18 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "Meter" has a relative path.
Call Stack (most recent call first):
  CMakeLists.txt:270 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at flashlight/memory/CMakeLists.txt:3 (add_library):
  add_library cannot create target "Memory" because another target with the
  same name already exists.  The existing target is an interface library
  created in source directory "/home/erik/Projects/flashlight".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  CMakeLists.txt:273 (include)


CMake Warning (dev) at flashlight/memory/CMakeLists.txt:20 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "Memory" has a relative path.
Call Stack (most recent call first):
  CMakeLists.txt:273 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at flashlight/nn/CMakeLists.txt:43 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "NN" has a relative path.
Call Stack (most recent call first):
  CMakeLists.txt:276 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at flashlight/optim/CMakeLists.txt:22 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  An interface source of target "Optim" has a relative path.
Call Stack (most recent call first):
  CMakeLists.txt:279 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring incomplete, errors occurred!
```
</details>

**Comment by jacobkahn** [comment]
@erksch — this broke last night and was fixed in ebb83373ec047c96efac064a3bb871f867c38ca8. Thanks for flagging/sorry about that.

**Comment by erksch** [comment]
No problem! Thanks for your reply!
